tutorial_R/raincloud_tutorial_r.Rmd

```diff
@@ -12,6 +12,19 @@ knitr::opts_chunk$set(fig.width=6, fig.height=3, fig.path='figs/',
                       echo=FALSE, warning=FALSE, message=FALSE)
 ```
 
+## Package dependencies
+
+Make sure we have the packages we need, and install them if they are missing.
+
+```{r package setup, include = TRUE, echo = TRUE}
+packages <- c("ggplot2", "dplyr", "lavaan", "plyr", "cowplot", "rmarkdown", 
+              "readr", "caTools", "bitops")
+
+if (length(setdiff(packages, rownames(installed.packages()))) > 0) {
+  install.packages(setdiff(packages, rownames(installed.packages())))  
+}
+```
```

Comment:
I think this chunk should go in the `R_rainclouds.R` file. Is there a strong reason to keep them out of it?

I do understand that being in the Rmarkdown file makes this step clearer, but I think it make the .R code more useful to have them there. Happy to be corrected as I'm not a regular R user!!

Reply:
I think there can be flexibility about where to put the calls to load packages (for instance, `R_rainclouds.R` loads in `ggplot2`), but adding code to install packages in R scripts that may be called externally from another place seems like bad practice.

With the current setup, you can run `R_rainclouds.R` without issue if you have the right packages installed, but you will need to load some of the other ones to run the examples. (I'm not sure how much of that is specific to the code for the examples vs. constructing a plot on one's own data.)

You could always make it into an R package to simplify installation... :)

Reply:
We shall have to have a longer conversation about why that's bad practice one day - but for now I'm totally fine with taking your word for it that this is what R users would expect so lets keep the code in the Rmarkdown file.

We SHOULD make it into an R package. I do *not* know how to do that so I'm going to make an issue to see if anyone wants to swing by to help out ✨